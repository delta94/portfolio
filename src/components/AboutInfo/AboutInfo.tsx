import clsx from 'clsx';
import React, { useContext } from 'react';
import { createUseStyles } from 'react-jss';
import { useMediaQuery } from 'react-responsive';
import { Button } from '..';
import { InformationsContext } from '../../App';
import { CustomTheme } from '../../theme';
import { MemoizedComponent } from '../../utils/custom-hook';
import { Informations } from '../../utils/informations';
import { SkillInformationsContainer } from '../SkillInformationsContainer/SkillInformationsContainer';

interface AboutInfoProps {}

const useStyles = createUseStyles((theme: CustomTheme) => ({
  mainContainer: {
    backgroundColor: theme.colors.darkerPrimaryColor,
    color: theme.colors.white,
    display: 'flex',
    padding: theme.spacing(2.5),
    borderRadius: 10,
    height: '100%',
    border: `1px solid ${theme.colors.secondaryColor}`,
  },
  descriptionContainer: {
    width: '100%',
    borderBottomColor: theme.colors.thirdColor,
    borderBottomWidth: 2,
    borderBottomStyle: 'solid',
    paddingBottom: theme.spacing(2),
  },
  descriptionContainerMobile: {
    borderBottomColor: theme.colors.thirdColor,
    borderBottomWidth: 2,
    borderBottomStyle: 'solid',
    borderRightColor: null,
    borderRightWidth: 0,
    borderRightStyle: null,
    paddingRight: theme.spacing(0),
  },
  description: {
    fontSize: 27,
  },
  descriptionMobile: {
    fontSize: 20,
  },
  button: {
    position: 'relative',
    top: 6,
    justifyContent: 'center',
    display: 'flex',
  },
  buttonMobile: {
    top: theme.spacing(0),
    marginBottom: theme.spacing(2.5),
    marginTop: theme.spacing(2.5),
  },
  skillsInfoContainer: {
    display: 'flex',
    flexDirection: 'column',
    width: '100%',
    justifyContent: 'space-around',
  },
  skillsInfoContainerMobile: {
    paddingLeft: theme.spacing(0),
    paddingTop: theme.spacing(1),
  },
  triangleContainer: {
    display: 'flex',
    alignItems: 'center',
  },
  triangle: {
    position: 'relative',
    right: theme.spacing(3.75),
    width: 0,
    height: 0,
    borderTop: `10px solid transparent`,
    borderBottom: `10px solid transparent`,
    borderRight: `10px solid ${theme.colors.secondaryColor}`,
  },
  triangleMobile: {
    left: theme.spacing(21.5),
    right: theme.spacing(0),
    top: theme.spacing(-4.375),
    transform: 'rotate(90deg)',
  },
  triangleContainerMobile: {
    alignSelf: 'end',
  },
  descriptionParentContainer: { display: 'flex', width: '100%', flexDirection: 'column' },
  descriptionParentContainerMobile: { flexDirection: 'column' },
}));

const AboutInfo: React.FC<AboutInfoProps> = () => {
  const classes: Record<
    | 'mainContainer'
    | 'descriptionContainer'
    | 'descriptionParentContainer'
    | 'descriptionParentContainerMobile'
    | 'triangle'
    | 'description'
    | 'descriptionContainerMobile'
    | 'button'
    | 'skillsInfoContainer'
    | 'triangleMobile'
    | 'triangleContainer'
    | 'buttonMobile'
    | 'triangleContainerMobile'
    | 'skillsInfoContainerMobile'
    | 'descriptionMobile',
    string
  > = useStyles();

  const isMobile: boolean = useMediaQuery({ maxWidth: 780 });
  const isSmallerDevices: boolean = useMediaQuery({ maxWidth: 480 });
  const context: Informations | undefined = useContext<Informations | undefined>(InformationsContext);

  const onClick = () =>
    window.open(
      'https://docs.google.com/document/d/16Ko4S8n50fedwi4re2K-djKxe2zGrRRWKuXdtko7jow/edit?usp=sharing',
      '_blank',
    );

  return (
    <div className={classes.mainContainer}>
      {MemoizedComponent(
        <div className={clsx([classes.triangleContainer, isMobile && classes.triangleContainerMobile])}>
          <div className={clsx([classes.triangle, isMobile && classes.triangleMobile])} />
        </div>,
        isMobile,
      )}
      {MemoizedComponent(
        <div
          className={clsx([classes.descriptionParentContainer, isMobile && classes.descriptionParentContainerMobile])}
        >
          <div className={clsx([classes.descriptionContainer, isMobile && classes.descriptionContainerMobile])}>
            <div className={clsx([classes.description, isSmallerDevices && classes.descriptionMobile])}>
              I am Shaikh Zeeshan Murshed, Full stack web developer from Montreal, Canada. I have experience in making
              Website. Also I am a hobbyist photographer.
            </div>
            <div className={clsx([classes.button, isMobile && classes.buttonMobile])}>
              <Button onClick={onClick}>Download My CV</Button>
            </div>
          </div>
          <div className={clsx([classes.skillsInfoContainer, isMobile && classes.skillsInfoContainerMobile])}>
            {context?.skillsInformation && (
              <SkillInformationsContainer skillsInformations={context?.skillsInformation} />
            )}
          </div>
        </div>,
        onClick,
      )}
    </div>
  );
};

export { AboutInfo };
