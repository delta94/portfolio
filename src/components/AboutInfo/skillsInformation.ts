import { theme } from '../../theme';

export interface SkillInformationTypes {
  name?: string;
  progress?: number;
  backgroundColor?: string;
}

const skillsInformation: SkillInformationTypes[] = [
  { name: 'React', progress: 80, backgroundColor: theme.colors.secondaryColor },
  { name: 'Typescript', progress: 75, backgroundColor: theme.colors.blue },
  { name: 'Redux', progress: 75, backgroundColor: theme.colors.thirdColor },
  { name: 'Photography', progress: 70, backgroundColor: theme.colors.lighterPrimaryColor },
];

export { skillsInformation };
