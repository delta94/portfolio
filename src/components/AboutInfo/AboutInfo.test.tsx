import { render } from '@testing-library/react';
import React from 'react';
import { TestContextProvider } from '../../TestContextProvider';
import { AboutInfo } from './AboutInfo';

test('AboutInfo renders correctly', () => {
  const { asFragment } = render(
    <TestContextProvider>
      <AboutInfo />
    </TestContextProvider>,
  );
  expect(asFragment()).toMatchSnapshot();
});
