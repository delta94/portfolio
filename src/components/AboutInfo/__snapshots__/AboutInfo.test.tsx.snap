// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AboutInfo renders correctly 1`] = `
<DocumentFragment>
  <div
    class="mainContainer-0-2-1"
  >
    <div
      class="triangleContainer-0-2-8"
    >
      <div
        class="triangle-0-2-9"
      />
    </div>
    <div
      class="descriptionContainer-0-2-2"
    >
      <div
        class="description-0-2-3"
      >
        I am Shaikh Zeeshan Murshed, Front-End developer from Montreal, Canada. I have experience in making Website. Also I am a hobbyist photographer.
      </div>
      <div
        class="button-0-2-5"
      >
        <button
          class="mainContainer-0-2-12"
          style="transform: perspective(600px) rotateX(0deg) rotateY(0deg) scale(1);"
        >
          Download My CV
        </button>
      </div>
    </div>
    <div
      class="skillsInfoContainer-0-2-7"
    />
  </div>
</DocumentFragment>
`;
